### PR TITLE
Add platforms annotation to kaniko task

### PR DESCRIPTION
**Pull request by barthy1**
# Changes

Annotation about `linux/amd64`,`linux/s390x`, and `linux/ppc64le` platforms was added to the latest versions of the kaniko task, as well as information which custom images to use for `linux/s390x` and `linux/ppc64le` platforms.
`JQ_IMAGE` parameter is introduced with default value from the existing task to be able to use custom image for non-Intel architecture.

The changes were tested for `linux/amd64`,`linux/s390x`, and `linux/ppc64le` platforms.

# Submitter Checklist

These are the criteria that every PR should meet, please check them off as you
review them:

- [x] Follows the [authoring recommendations](https://github.com/tektoncd/catalog/blob/main/recommendations.md)
- [x] Includes [docs](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if user facing)
- [ ] Includes [tests](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if functionality of task changed or new task added)
- [x] Commit messages follow [commit message best practices](https://github.com/tektoncd/community/blob/master/standards.md#commit-messages)
- [ ] Complies with [Catalog Organization TEP][TEP], see [example]. **Note** [An issue has been filed to automate this validation][validation]
  - [ ] File path follows  `<kind>/<name>/<version>/name.yaml`
  - [ ] Has `README.md` at `<kind>/<name>/<version>/README.md`
  - [ ] Has mandatory `metadata.labels` - `app.kubernetes.io/version` the same as the `<version>` of the resource
  - [ ] Has mandatory `metadata.annotations` `tekton.dev/pipelines.minVersion`
  - [ ] mandatory `spec.description` follows the convention

          ```

          spec:
            description: >-
              one line summary of the resource

              Paragraph(s) to describe the resource.
          ```

_See [the contribution guide](https://github.com/tektoncd/catalog/blob/master/CONTRIBUTING.md)
for more details._

---

[TEP]: https://github.com/tektoncd/community/blob/master/teps/0003-tekton-catalog-organization.md
[example]: https://github.com/tektoncd/catalog/tree/master/task/git-clone/0.1
[validation]:  https://github.com/tektoncd/catalog/issues/413

### CODE REVIEW

File: task/kaniko/0.4/README.md [hunk 1]
```diff
@@ -52,6 +53,15 @@ the docker `config.json`.
...
 configuration file in a key called `config.json`.
 
+## Platforms
+
+The Task can be run on `linux/amd64`, `linux/s390x` and `linux/ppc64le` platforms.
+
+For `linux/s390x` platform specify `BUILDER_IMAGE` parameter with `gcr.io/kaniko-project/executor:s390x-9ed158c1f63a059cde4fd5f8b95af51d452d9aa7`
+value and `JQ_IMAGE` parameter with `ibmcom/jq-s390x:latest` value  in TaskRun or PipelineRun.
+
+For `linux/ppc64le` platform specify `JQ_IMAGE` parameter with `ibmcom/jq-ppc64le:latest` value  in TaskRun or PipelineRun.
```
[the remark this answers is not in the document]
Comment (barthy1) [reply task/kaniko/0.4/README.md:63]: @vinamra28 would it be ok for you to resolve this issue step be step?
I mean first step would be just document the differences between platforms in documentation, next step to either somehow generate multi-arch image or define task differently.

[the remark this answers is not in the document]
Comment (vinamra28) [reply task/kaniko/0.4/README.md:63]: yeah will be good for first iteration :+1:

**Comment by vdemeester** [comment]
```
task check-name-match has failed: "step-check-name" exited with code 2 (image: "docker-pullable://alpine@sha256:e1c082e3d3c45cccac829840a25941e679c25d438cc8412c2fa221cf1a824e6a"); for logs run: kubectl -n tekton-ci logs pull-catalog-catlin-run-ssmzq-check-name-match-f85nx-pod-8nl2t -c step-check-name

[check-name] + set -o pipefail
[check-name] + printf passed
[check-name] + '[[' '/test pull-catalog-catlin-lint
[check-name]
[check-name] ' '=='  ]]
[check-name] /tekton/scripts/script-0-8rgz4: line 19: syntax error: unexpected "|"
```

🤔

**Pull request merged**